### DISCUSSION TITLE
Chunked Upload with Cloudinary autogenerated public ids

### DISCUSSION THREAD

**Discussion by maxsalven** [rpldy/react-uploady#236] [Q&A] [answered]
I'm trying to use ChunkedUploady with a large video file on Cloudinary.

I'm using an unsigned upload with an upload preset.

I have Cloudinary set to auto generate a random public id, I don't want the client to be allowed to specify the id.

The first chunk uploads successfully, but then the second chunk has an error:
`message: "Missing required parameter - public_id"`

Presumably the second chunk upload needs to be aware of the public_id that cloudinary generated when it received the first chunk?

**Comment by maxsalven** [comment, marked as answer]
I think I found a solution, using this help center article (I couldn't see any mention of this in the Cloudinary docs but that may well be user error):

https://support.cloudinary.com/hc/en-us/articles/208263735-Guidelines-for-self-implementing-chunked-upload-to-Cloudinary

"You must include the `X-Unique-Upload-Id` header and make sure it's identical (and unique) for all parts."

I utilized `useRequestPreSend` for this:

```
useRequestPreSend(() => {
  return {
      options: {
        destination: {
          headers: {
            "X-Unique-Upload-Id": uuid(),
          },
        },
      },
    };
});
```

Does this seem like an appropriate solution?

**Comment by yoavniran** [comment]
Hi @maxsalven that's exactly the solution. I was sure I documented this more broadly.

It is being used exactly like that in [one of the stories in the repo](https://github.com/rpldy/react-uploady/blob/master/packages/ui/chunked-uploady/ChunkedUploady.stories.js#L41).

I will add it to the [Cloudinary Guide](https://github.com/rpldy/react-uploady/blob/master/guides/Cloudinary.md) as well. 

Thanks!
